src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -27,16 +28,28 @@ async function boot(userOptions?: Partial<BlazorOptions>): Promise<void> {
   options.reconnectionHandler = options.reconnectionHandler || window['Blazor'].defaultReconnectionHandler;
   logger.log(LogLevel.Information, 'Starting up blazor server-side application.');
 
-  const initialConnection = await initializeConnection(options, logger);
-  const circuit = await startCircuit(initialConnection);
+  const components = discoverComponents(document);
+  const circuit = new CircuitDescriptor(components);
+
+  const initialConnection = await initializeConnection(options, logger, circuit);
+  const circuitStarted = await circuit.startCircuit(initialConnection);
+  if (!circuitStarted) {
+    logger.log(LogLevel.Error, 'Failed to start the circuit.');
+    return;
+  }
 
   const reconnect = async (existingConnection?: signalR.HubConnection): Promise<boolean> => {
     if (renderingFailed) {
       // We can't reconnect after a failure, so exit early.
       return false;
     }
 
-    const reconnection = existingConnection || await initializeConnection(options, logger);
+    const reconnection = existingConnection || await initializeConnection(options, logger, circuit);
+    if (reconnection.state !== signalR.HubConnectionState.Connected) {
+      logger.log(LogLevel.Information, 'Reconnection attempt failed. Unable to connect to the server.');
+      return false;
+    }
```

Comment:
This `if` block was removed by some changes that are pretty recent by @NTaylorMullen - signalR actually throws for you so this would never run.

Reply:
My bad, bad merge